WordPress/wordpress-playground
# Remove DOMParser from Blueprints: installPlugin and installTheme

## What?



The goal of this pull request is to remove the use of any browser-specific API such as `DOMParser` from the Blueprints package. This allows all Blueprint steps to run on the browser and server side.

## Why?



Currently, the steps `installPlugin` and `installTheme` can only be run with the Playground in the browser, not on server side with `wp-now` and Node.js, because they use the `asDOM` helper function which depends on the `DOMParser` class only available in the browser.

Related discussion:

- #379 

The initial idea was to introduce an "isomorphic DOM" library that exports native DOM API for the browser and [jsdom](https://github.com/jsdom/jsdom) for Node.js. That would have allowed the existing implementation of `installPlugin` and `installTheme` to work as is on server side.

However, after weighing the pros and cons, it was decided that it's simpler to maintain if we rewrite these steps to perform their actions without using any DOM operations.

## How?



- Rewrite the Blueprint steps `installPlugin` and `installTheme` to use Playground and PHP WASM API.
- Remove the `asDOM` helper function.

## Testing Instructions


1. Check out the branch.
2. Run `npx nx test playground-blueprints`
